### PR TITLE
Editorial: rename event path's item and target

### LINKED ISSUE #645: Editorial: improve clarity in event dispatch model
In https://bugzilla.mozilla.org/show_bug.cgi?id=1449560#c5 @bzbarsky wrote:

> In particular, the fact that the "target" in the dispatch algorithm becomes the "item" in the event path structs.

I'd welcome ideas on how to improve this. I agree it's a bit of a mess, but I also don't quite see how to make things substantially better. We could rename to "item" to "current target" perhaps, but I'm not sure if that helps or hurts things.

> I _really_ hope "the last tuple in event’s path whose target is non-null" is in fact the thing that we are thinking of as the target of the event... Is it?

It's the outermost target. The innermost target (the one passed to dispatch) will be the first tuple in event's path whose target is non-null.

### PR DESCRIPTION
**Pull request by annevk**
Fixes #645.

***
<a href="https://whatpr.org/dom/753.html" title="Last updated on May 17, 2019, 10:06 AM UTC (d229069)">Preview</a> | <a href="https://whatpr.org/dom/753/98564fc...d229069.html" title="Last updated on May 17, 2019, 10:06 AM UTC (d229069)">Diff</a>

### CODE REVIEW

**Comment by annevk** [comment]
@bzbarsky @rniwa how does this look?

I could also rename _parent_ to _currentTarget_ or some such, as I think was suggested in #645, but it's not totally clear to me that adds clarity. I might be in a bit too deep though.

File: dom.bs [hunk 1]
```diff
@@ -483,10 +483,10 @@ empty list.
...
 
 <p>An <a>event</a> has an associated <dfn export for=Event>path</dfn>. A <a for=Event>path</a> is a
-<a for=/>list</a> of <a for=/>structs</a>. Each <a for=/>struct</a> consists of an
-<dfn for=Event/path>item</dfn> (an {{EventTarget}} object), an
-<dfn for=Event/path>item-in-shadow-tree</dfn> (a boolean), a <dfn for=Event/path>target</dfn> (a
-<a>potential event target</a>), a
+<a for=/>list</a> of <a for=/>structs</a>. Each <a for=/>struct</a> consists of a
+<dfn for=Event/path>current target</dfn> (an {{EventTarget}} object), a
+<dfn for=Event/path>current-target-in-shadow-tree</dfn> (a boolean), a
+<dfn for=Event/path>target</dfn> (a <a>potential event target</a>), a
```
Comment (rniwa) [inline dom.bs]: Maybe we can call this one *dispatch target* to further distinguish itself from other kinds of targets?

File: dom.bs [hunk 2]
```diff
@@ -613,7 +614,7 @@ steps:
    <li><p>If <var>path</var>[<var>index</var>]'s <a for=Event/path>root-of-closed-tree</a> is true,
    then increase <var>currentTargetHiddenSubtreeLevel</var> by 1.
 
-   <li><p>If <var>path</var>[<var>index</var>]'s <a for=Event/path>item</a> is
+   <li><p>If <var>path</var>[<var>index</var>]'s <a for=Event/path>current target</a> is
```
Comment (rniwa) [inline dom.bs]: *current target* being different form *currentTarget* is super confusing here...
Perhaps we should call *current target* in the struct *invocation target* instead??

File: dom.bs (hunk 1 above)
Comment (annevk) [reply dom.bs]: Okay, so this is either the target passed into the dispatch algorithm or the target as adjusted for shadow tree encapsulation (some host). Does originating target reflect that well?

Comment (rniwa) [reply dom.bs]: Hm... maybe shadow-adjusted originating target? I think the key here is convey the semantics that the event looks as though it originated from that node in scripts.

Or we could call it *shadow adjusted target* or *visible originating target*. Or maybe even *embarkment target* or *source target*?

**Comment by annevk** [comment]
Updated the PR to use invocation and shadow-adjusted target as terms.

**Comment by annevk** [comment]
@smaug---- you might want to take a look as well.

File: dom.bs [hunk 3]
```diff
@@ -483,10 +483,10 @@ empty list.
 {{TouchEvent}} interface and related interfaces. [[TOUCH-EVENTS]]
 
 <p>An <a>event</a> has an associated <dfn export for=Event>path</dfn>. A <a for=Event>path</a> is a
-<a for=/>list</a> of <a for=/>structs</a>. Each <a for=/>struct</a> consists of an
-<dfn for=Event/path>item</dfn> (an {{EventTarget}} object), an
-<dfn for=Event/path>item-in-shadow-tree</dfn> (a boolean), a <dfn for=Event/path>target</dfn> (a
-<a>potential event target</a>), a
+<a for=/>list</a> of <a for=/>structs</a>. Each <a for=/>struct</a> consists of a
+<dfn for=Event/path>invocation target</dfn> (an {{EventTarget}} object), a
+<dfn for=Event/path>current-target-in-shadow-tree</dfn> (a boolean), a
```
Comment (smaug----) [inline dom.bs]: current target has very specific meaning, so why do we use it here too?
Shouldn't it be invocation-target-in-shadow-tree?

Comment (annevk) [reply dom.bs]: You're right, thanks!

**Comment by annevk** [comment]
@rniwa @smaug---- I think it's all good now. If this event terminology looks good hopefully it can stick around for a while.

**Comment by rniwa** [comment]
The new terminology reads a lot better to me.

**Pull request merged**